tests/behat/features/bootstrap/LibContext.php

```diff
@@ -0,0 +1,1399 @@
+   *
+   * @Then I save (a) screenshot
+   */
+  public function saveScreenshot() {
+    // Initially check if we are taking screenshots at all. The setting is in the behat.yml file.
+    if (!$this->takeScreenDumps) {
+      print_r("Due to screendump setting in behat.yml, screendumps are currently not being saved.\n");
+      return;
+    }
+
+    // Setup folders and make sure the folders exists.
+    $screenShotDir = $this->verbose->ScrShotDir;
+    $featureFolder = "";
+    if ($this->verbose->ScrShotUseFeatureFolder) {
+      $featureFolder = preg_replace('/\W/', '', ucwords(strtolower($this->currentFeature))) ;
+      if (!file_exists($screenShotDir . $featureFolder)) {
+        mkdir($screenShotDir . $featureFolder);
+      }
+      // Add the slash to make the following code work in all circumstances.
+      $featureFolder = $featureFolder . "/";
+    }
+
+    // Setup filename and make sure it is unique, by adding a postfix (simple number).
+    $fileName = $screenShotDir . $featureFolder
+      . preg_replace('/\W/', '', ucwords(strtolower($this->currentScenario)));
+    $fileNamePostfix = "";
+    while (file_exists($fileName . $fileNamePostfix . '.png')) {
+      $fileNamePostfix++;
+    }
+    $fileName = $fileName . $fileNamePostfix . '.png';
+
+    // Log the filename of the screenshot to notify the user.
+    print_r("Screenshot in: " . $fileName . "\n");
+
+    // Now find the actual height of the shown page.
+    $height = $this->minkContext->getSession()->evaluateScript("return document.body.scrollHeight;");
+    // Save the screenshot.
+    $this->minkContext->getSession()
+          ->getDriver()
+          ->resizeWindow(1280, $height, 'current');
+    file_put_contents($fileName, $this->minkContext->getSession()->getDriver()->getScreenshot());
+  }
+
+  /**
+   * Step
+   *
+   * @Given I accept cookies
+   *
+   * @throws Exception
+   *   Checks if cookie acceptance is shown, and accepts it if it is.
+   */
+  public function acceptCookiesMinimizeAskLibrarianOverlay() {
+    // We use the searchPage-instance to deal with cookies.
+    $this->check($this->searchPage->acceptCookiesMinimizeAskLibrarianOverlay(), $this->searchPage->getMessages());
+  }
+
+  /**
+   * General function to check the outcome of a function and handle the result
+   *
+   * @param string $result
+   *   If non-empty an exception is thrown.
+   * @param string $msg
+   *   The message to show together with the exception.
+   *
+   * @throws Exception
+   *   In case of error.
+   */
+  public function check($result, $msg = '') {
+    // Log messages if we have any.
+    if ($msg !== '') {
+      print_r($msg);
+    }
+    // Fail if we have a non-empty string.
+    if ($result !== "") {
+      throw new Exception($result);
+    }
+  }
+
+
+  /**
+   * Step
+   *
+   * @Then I check if the right number of search results are shown
+   *
+   * The methods "I use facets to reduce..." sets $this->>expectedResultsCount and must
```

Comment:
Move to after DocBlock Summary as DocBlock description

Reply:
Fixed.